Components/TEStorageHeart.cs

```diff
@@ -626,27 +680,38 @@ public Item TryWithdraw(Item lookFor, bool keepOneIfFavorite, bool toInventory =
 				return;
 			}
 
-			bool didSomething = false;
+			try {
+				bool didSomething = false;
 
-			foreach (Item item in GetStorageUnits().OfType<TEStorageUnit>().SelectMany(s => s.GetItems())) {
-				//Filter out air items and Unloaded Items (their data might belong to the mod they're from)
-				if (item is null || item.IsAir || item.ModItem is UnloadedItem)
-					continue;
+				HashSet<int> typesToRefresh = new();
 
-				if (Item_globalItems.GetValue(item) is not Instanced<GlobalItem>[] globalItems || globalItems.Length == 0)
-					continue;
+				foreach (Item item in GetStorageUnits().OfType<TEStorageUnit>().SelectMany(s => s.GetItems())) {
+					//Filter out air items and Unloaded Items (their data might belong to the mod they're from)
+					if (item is null || item.IsAir || item.ModItem is UnloadedItem)
+						continue;
+
+					if (Item_globalItems.GetValue(item) is not Ref<GlobalItem>[] globalItems || globalItems.Length == 0)
```

Comment:
i'll need to examine this code again; just leaving this comment here as a reminder